`master` is broken from incorrect call to `typeToTypeNode`

```
src/services/codefixes/fixReturnTypeInAsyncFunction.ts:49:42 - error TS2554: Expected 3-4 arguments, but got 1.

49         const promisedTypeNode = checker.typeToTypeNode(promisedType);
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~

  built/local/compiler.d.ts:3099:36
    3099         typeToTypeNode(type: Type, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined): TypeNode | undefined;
                                            ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    An argument for 'enclosingDeclaration' was not provided.


Found 1 error.
```

--- a/src/services/codefixes/fixReturnTypeInAsyncFunction.ts
+++ b/src/services/codefixes/fixReturnTypeInAsyncFunction.ts
@@ -52,7 +52,7 @@ namespace ts.codefix {
 
         const returnType = checker.getTypeFromTypeNode(returnTypeNode);
         const promisedType = checker.getAwaitedType(returnType) || checker.getVoidType();
-        const promisedTypeNode = checker.typeToTypeNode(promisedType, /*enclosingDeclaration*/ func, /*flags*/ undefined);
+        const promisedTypeNode = checker.typeToTypeNode(promisedType, /*enclosingDeclaration*/ returnTypeNode, /*flags*/ undefined);
         if (promisedTypeNode) {
             return { returnTypeNode, returnType, promisedTypeNode, promisedType };
         }

--- a/src/services/codefixes/fixReturnTypeInAsyncFunction.ts
+++ b/src/services/codefixes/fixReturnTypeInAsyncFunction.ts
@@ -39,14 +39,20 @@ namespace ts.codefix {
     });
 
     function getInfo(sourceFile: SourceFile, checker: TypeChecker, pos: number): Info | undefined {
-        const returnTypeNode = getReturnTypeNode(sourceFile, pos);
+        if (isInJSFile(sourceFile)) {
+            return undefined;
+        }
+
+        const token = getTokenAtPosition(sourceFile, pos);
+        const func = findAncestor(token, isFunctionLikeDeclaration);
+        const returnTypeNode = func?.type;
         if (!returnTypeNode) {
             return undefined;
         }
 
         const returnType = checker.getTypeFromTypeNode(returnTypeNode);
         const promisedType = checker.getAwaitedType(returnType) || checker.getVoidType();
-        const promisedTypeNode = checker.typeToTypeNode(promisedType);
+        const promisedTypeNode = checker.typeToTypeNode(promisedType, /*enclosingDeclaration*/ func, /*flags*/ undefined);
         if (promisedTypeNode) {
             return { returnTypeNode, returnType, promisedTypeNode, promisedType };
         }
@@ -54,17 +60,5 @@ namespace ts.codefix {
 
     function doChange(changes: textChanges.ChangeTracker, sourceFile: SourceFile, returnTypeNode: TypeNode, promisedTypeNode: TypeNode): void {
         changes.replaceNode(sourceFile, returnTypeNode, createTypeReferenceNode("Promise", [promisedTypeNode]));
-    }
-
-    function getReturnTypeNode(sourceFile: SourceFile, pos: number): TypeNode | undefined {
-        if (isInJSFile(sourceFile)) {
-            return undefined;
-        }
-
-        const token = getTokenAtPosition(sourceFile, pos);
-        const parent = findAncestor(token, isFunctionLikeDeclaration);
-        if (parent?.type) {
-            return parent.type;
-        }
     }
 }

--- a/tests/cases/fourslash/codeFixReturnTypeInAsyncFunction18.ts
+++ b/tests/cases/fourslash/codeFixReturnTypeInAsyncFunction18.ts
@@ -1,0 +1,20 @@
+/// <reference path='fourslash.ts' />
+
+// @target: es2015
+////
+////interface A {}
+////export { A as PublicA };
+////async function foo(): A {
+////    return {}
+////}
+
+verify.codeFix({
+    index: 0,
+    description: [ts.Diagnostics.Replace_0_with_Promise_1.message, "A", "A"],
+    newFileContent: `
+interface A {}
+export { A as PublicA };
+async function foo(): Promise<A> {
+    return {}
+}`
+});

